js/core/webidl-contiguous.js

```diff
@@ -640,9 +640,12 @@ define(
 
                     case "enum":
                         name = defn.name;
-                        defn.values.filter(function (v) { return v.type === undefined;})
-                                   .forEach(function(v) {
-                            v.dfn = findDfn(name, v, definitionMap, msg);
+                        defn.values.forEach(function(v,i) {
+                            if (v.type === undefined) {
```

Comment:
I don't understand why we can't filter here? 

Shouldn't this just:
```JS
defn.values
  .filter(function(item){return typeof item === "undefined"})
  // create the objects
  .map(function(item){ return { 
    toString: function() {return item;},
    dfn: findDfn(name, item, definitionMap, msg),
  })
  // reduce them into the defn
  .reduce(function(collector, next, index){  
       collector.values[i] = next;
       return collector;
   },  defn);
```

Or am I missing something? Is it that it needs to use the original index?